micropub.php

```diff
@@ -516,6 +526,25 @@ private static function mp_to_wp( $mf2 ) {
 		if ( isset( $args['post_name'] ) ) {
 			$args['post_name'] = sanitize_title( $args['post_name'] );
 		}
+		if ( ! isset( $props['post-status'] ) ) {
+			$args['post_status'] = get_option( 'micropub_default_post_status' );
+		} else {
+			//  According to the proposed specification these are the only two properties supported.
+			// https://indieweb.org/Micropub-extensions#Post_Status
+			// For now these are the only two we will support even though WordPress defaults to 8 and allows custom
+			// But makes it easy to change
+			$statuslist = array( array( 'published' ), array( 'draft' ) );
+			if ( ! in_array( $props['post-status'], $statuslist, true ) ) {
+				$args['post_status'] = get_option( 'micropub_default_post_status' );
```

Comment:
we should return a 400 error here, or whatever the micropub spec says to do, instead of silently ignoring the status they requested.

Reply:
As I sort of alluded to in chat, https://www.w3.org/TR/micropub/#unrecognized-properties

"If the request includes properties that the server does not recognize, it MUST ignore unrecognized properties and create the post with the values that are recognized."

So, if the server doesn't recognize the status, it should ignore it by this statement. Since post-status is a proposed extension to Micropub, and not part of the specification...

https://indieweb.org/Micropub-extensions#Post_Status

Reply:
Thinking it out, I get the point here.